File: front-end/Norlan-Svelte-TypeScript-master/src/lib/models/Messaggio.ts
/*
Modello di dominio per i messaggi scambiati all'interno della piattaforma.
Gestisce le informazioni su mittente, destinatario, contenuto testuale e stato di lettura.
*/
export interface MessaggioData {
	idMessaggio: number;
	idMittente: number;
	nomeMittente: string;
	idDestinatario: number;
	testo: string;
	timestampInvio: string;
	letto: boolean;
}

export class Messaggio {
	idMessaggio: number;
	idMittente: number;
	nomeMittente: string;
	idDestinatario: number;
	testo: string;
	timestampInvio: string;
	letto: boolean;

	constructor(data: MessaggioData) {
		this.idMessaggio = data.idMessaggio;
		this.idMittente = data.idMittente;
		this.nomeMittente = data.nomeMittente;
		this.idDestinatario = data.idDestinatario;
		this.testo = data.testo;
		this.timestampInvio = data.timestampInvio;
		this.letto = data.letto;
	}
}

File: back-end/norlan-client-portal-master/src/main/java/it/norlan/clientportal/model/Messaggio.java
package it.norlan.clientportal.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

/**
 * Entità JPA che modella i messaggi della chat interna.
 * Memorizza il testo della conversazione, lo stato di lettura e traccia le relazioni
 * (Many-to-One) sia con l'utente mittente che con il destinatario.
 */

@Getter @Setter
@NoArgsConstructor @AllArgsConstructor
@Entity
@Table(name = "messaggio")
public class Messaggio {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer idMessaggio;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_mittente", nullable = false)
    private Utente mittente;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_destinatario", nullable = false)
    private Utente destinatario;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String testo;

    @Column(name = "timestamp_invio")
    private LocalDateTime timestampInvio = LocalDateTime.now();

    @Column(name = "letto")
    private Boolean letto = false;
}
